File: services/main/frontend_old/lib/src/components/workspace_image_badge/workspace_image_badge_component.dart
import 'package:angular/angular.dart';
import '../../model/core.dart';

@Component(
  selector: 'workspace-image-badge',
  templateUrl: 'workspace_image_badge_component.html',
  styleUrls: const ['workspace_image_badge_component.css'],
  directives: const [coreDirectives]
)
class WorkspaceImageBadgeComponent {

  @Input()
  WorkspaceImage image;

  String get name => image.imageName.split(':')[0];
}

File: services/main/frontend/src/app/core/components/workspace-image-badge/workspace-image-badge.component.ts
import { Component, Input, OnInit } from '@angular/core';
import { WorkspaceImage } from '../../models/workspace-image';

@Component({
  selector: 'cc-workspace-image-badge',
  templateUrl: './workspace-image-badge.component.html',
  styleUrls: ['./workspace-image-badge.component.css']
})
export class WorkspaceImageBadgeComponent {

  @Input()
  image: WorkspaceImage;
}
